src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1619,7 +1619,16 @@ func handleImageGuestPullBlockVolume(c *Container, virtualVolumeInfo *types.Kata
 		case podmanAnnotations.ContainerType:
 			image_ref = container_annotations["io.kubernetes.cri-o.ImageName"]
 		default:
-			image_ref = ""
+			// There are cases, like when using nerdctl, where the criContainerType
+			// will never be set, leading to this code path.
+			//
+			// nerdctl also doesn't set any mechanism for automatically setting the
+			// image, but as part of it's v2.0.0 release it allows the user to set
+			// any kind of OCI annotation, which we can take advantage of and use.
+			//
+			// With this in mind, let's "fallback" to the default k8s cri image-name
+			// annotation, as documented on our image-pull documentation.
+			image_ref = container_annotations["io.kubernetes.cri.image-name"]
```

Comment:
Optional idea - do you think having `"io.kubernetes.cri.image-name"` and `"io.kubernetes.cri-o.ImageName"` as const at the top would be helpful?

Reply:
Done, thanks @stevenhorsman!